matrixio-core.c

```diff
@@ -77,7 +77,8 @@ int matrixio_hw_reg_read(void *context, unsigned int reg, unsigned int *val)
 	hw_addr->burst = 0;
 	hw_addr->readnwrite = 1;
 
-	ret = matrixio_spi_transfer(matrixio->spi, send_buf, (uint8_t *)recv_buf, 4);
+	ret = matrixio_spi_transfer(matrixio->spi, send_buf,
+				    (uint8_t *)recv_buf, 4);
```

Comment:
Wondering... Why send buf 4 and recv buf 2? Is the address in send buff? (learning)

Reply:
Due to SPI intrinsics..  when one bit is written, one bit is read (in the same clock event) .  The internal FSM in the FPGA don't have noting to write when the first 2 bytes was written.